chia/full_node/pending_tx_cache.py

```diff
@@ -40,3 +47,68 @@ def drain(self) -> Dict[bytes32, MempoolItem]:
 
     def cost(self) -> int:
         return self._cache_cost
+
+
+@dataclass
+class PendingTxCache:
+    _cache_max_total_cost: int
+    _cache_max_size: int = 3000
+    _cache_cost: int = field(default=0, init=False)
+    _txs: Dict[bytes32, MempoolItem] = field(default_factory=dict, init=False)
+    _by_height: SortedDict[uint32, Dict[bytes32, MempoolItem]] = field(default_factory=SortedDict, init=False)
+
+    def get(self, bundle_name: bytes32) -> Optional[MempoolItem]:
+        return self._txs.get(bundle_name, None)
+
+    def add(self, item: MempoolItem) -> None:
+        """
+        Adds SpendBundles that are not yet valid because of a height assertion.
+        They will be re-tried once their height requirement is satisfied
+        """
+        assert item.assert_height is not None
+
+        name = item.name
+
+        if name in self._txs:
+            return None
+
+        self._txs[name] = item
+        self._cache_cost += item.cost
+        self._by_height.setdefault(item.assert_height, {})[name] = item
+
+        while self._cache_cost > self._cache_max_total_cost or len(self._txs) > self._cache_max_size:
+
+            # we start removing items with the highest assert_height first
```

Comment:
It may not be this simple, but is this what a perfectly greedy farmer would do?  Wouldn't they keep the ones with the higher fee around?  I don't think that the probability of them winning a block now versus later is any different, right?

Reply:
you're right. This is likely not the optimal eviction policy. I don't think its worse than what we have now though (which is FIFO). I imagine we'll revisit this once we make the mempool more sophisticated. The balance is that there's a cost to keeping these transactions in memory, and transactions that won't be valid for a very long time are more expensive to keep in RAM, because they will use RAM for longer.

But the fee-per-cost probably should be taken into account for a better strategy.